include/dsn/c/api_layer1.h

```diff
@@ -785,6 +785,9 @@ extern DSN_API void          dsn_msg_write_next(
 /*! commit the write buffer after the message content is written with the real written size */
 extern DSN_API void          dsn_msg_write_commit(dsn_message_t msg, size_t size);
 
+/*! append a buffer to the message, the "data" can be reinterpreted cast to (blob*)  */
+extern DSN_API void          dsn_msg_write_append(dsn_message_t msg, dsn_blob_t data);
```

Comment:
well, now I will discard the optimization and leave it to future.

On Fri, Jul 29, 2016 at 5:31 PM, Zhenyu Guo <notifications@github.com>
wrote:

> In include/dsn/c/api_layer1.h
> <https://github.com/imzhenyu/rDSN/pull/499#discussion_r72765645>:
>
> > @@ -785,6 +785,9 @@ extern DSN_API void          dsn_msg_write_next(
> >  /*! commit the write buffer after the message content is written with the real written size */
> >  extern DSN_API void          dsn_msg_write_commit(dsn_message_t msg, size_t size);
> >
> > +/*! append a buffer to the message, the "data" can be reinterpreted cast to (blob*)  */
> > +extern DSN_API void          dsn_msg_write_append(dsn_message_t msg, dsn_blob_t data);
>
> This is what is ugly about C interface, unfortunately. That is why in
> certain cases I would prefere ref_counter or instrusive ptr instead of
> shared-ptr:-( I'll think about this though to see how we can enable this
> more elegantly.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/imzhenyu/rDSN/pull/499/files/ea230134983ab865c8bb7b6e2110776c7d14b9ac#r72765645>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAgQPbwWAMkLxrIkYmRDBcyBPWZX73lbks5qach4gaJpZM4JX4Dq>
> .
>